resqpy/surface/_surface.py

```diff
@@ -918,6 +918,45 @@ def axial_edge_crossings(self, axis, value = 0.0):
 
         return ep
 
+    def resampled_surface(self, title = None):
+        """Creates a new triangulated set which is a resampled version of the current triangulated set. Each existing triangle in the tset is divided equally into 4 new triangles.
+           
+        arguments:
+            title (str): a new title for the output triangulated set, if None the title will have the same title as the input triangulated set
+        
+        returns:
+            resqpy.surface.Surface object, with extra_metadata ('resampled from surface': <uuid>), where uuid is the origin surface uuid
+        """
```

Comment:
Worth adding a comment about the edge-based alternative possible implementation, in case we need to optimise at a later date.